### PR TITLE
Ian/api updates

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Change description
A number of changes have been made to consolidate assessment exporting with the new API.

* Removed separate functions to export charts and assets
* consolidated exporting into lens. While it's possible to export results without using Lens, it isn't easy.
* Removed charts from exporting. Currently there is never a need to export a chart separately from the html report.
* Removed report name from create_reports argument. Report name is automatically created

Minor changes:
* fairnessBase had some leftover vestigial arguments
* simplified how reports were saved as html (now as_html argument. Switched to automatic eval of file type)

Future to-do
* All reports are "fairness" right now. We will need a future state where the type of risk lens is reporting on is explicitly stated.

### CODE REVIEW

File: credoai/integration.py [hunk 1]
```diff
@@ -244,48 +243,38 @@ def record_metrics_from_dict(metrics, **metadata):
...
-    post_figure(credo_id, figure_record)
+    prepared_results : list
+        prepared of prepared_results from credo_assessments. See lens.export_assessments for example
+    report : credo.reporting.NotebookReport, optional
+        report to optionally include with assessments, by default None
+    assessed_at : str, optional
+        date when assessments were created, by default None
+    """    
+    # prepare assessments
+    assessment_records = [record_metrics(r) for r in prepared_results]
```
Comment (IanAtCredo) [inline credoai/integration.py:259]: @aproxacs Here is where the API payload is created.

File: credoai/utils/credo_api_utils.py [hunk 2]
```diff
@@ -153,46 +156,21 @@ def get_model_project_by_name(project_name):
...
-        Model Record object, see credo.integration.MutliRecord
-    """
-    end_point = get_end_point(f"models/{model_id}/relationships/metrics")
-    return submit_request('patch', end_point, data=model_record.jsonify(), headers={"content-type": "application/vnd.api+json"})
-
-
-def post_figure(model_id, figure_record):
-    """Send a figure record object to Credo AI's Governance Platform
+def post_assessment(use_case_id, model_id, data):
+    end_point = get_end_point(f"use_cases/{use_case_id}/models/{model_id}/assessments")
```
Comment (IanAtCredo) [inline credoai/utils/credo_api_utils.py:168]: @aproxacs new api call to send assessment and get status.

**Review by amrasekh — APPROVED** [review]

File: credoai/lens.py [hunk 3]
```diff
@@ -191,49 +176,58 @@ def create_reports(self, report_name, export=False, display_results=False):
...
         final_report.create_report(self)
         # exporting
-        names = self.get_artifact_names()
-        name_for_save = f"{report_name}_model-{names['model']}_data-{names['dataset']}.html"
-        if isinstance(export, str):
-            final_report.write_notebook(
-                path.join(export, name_for_save), as_html=True)
-        elif export:
+        if export:
+                self.export_asessments(export, report=final_report)
```
Comment (amrasekh) [inline credoai/lens.py:183]: It seems `export_asessments` must be `export_assessments`
@IanAtCredo

Comment (IanAtCredo) [reply credoai/lens.py]: fixed

**Pull request merged**